packages/metal-jsx/src/otherProps.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+import { object } from 'metal';
+
+/**
+ * Returns props that are not used or declared in the component.
+ * @param {Component} component Instance of metal-jsx component
+ * @return {Object} Object containing props
+ */
+export default function otherProps(component) {
+	const removeKeys = [...component.getDataManager().getPropsInstance(component).getStateKeys(), 'key', 'ref'];
+
+	const retObj = object.mixin({}, component.props);
+
+	removeKeys.forEach(
```

Comment:
I'm not sure if this really needs to change, but I can see most of the core library mostly prefers the C-style `for` loop over `forEach`. I'm guessing for performance? (The overhead of a function invocation for `n` items).